builtin/credential/userpass/path_users.go

```diff
@@ -43,15 +42,32 @@ func pathUsers(b *backend) *framework.Path {
 		Callbacks: map[logical.Operation]framework.OperationFunc{
 			logical.DeleteOperation: b.pathUserDelete,
 			logical.ReadOperation:   b.pathUserRead,
-			logical.UpdateOperation:  b.pathUserWrite,
+			logical.UpdateOperation: b.pathUserWrite,
+			logical.CreateOperation: b.pathUserWrite,
 		},
 
+		ExistenceCheck: b.userExistenceCheck,
+
 		HelpSynopsis:    pathUserHelpSyn,
 		HelpDescription: pathUserHelpDesc,
 	}
 }
 
-func (b *backend) User(s logical.Storage, n string) (*UserEntry, error) {
+func (b *backend) userExistenceCheck(req *logical.Request, data *framework.FieldData) (bool, error) {
+	username := data.Get("username").(string)
+	if username == "" {
+		return false, fmt.Errorf("missing username")
+	}
+
+	userEntry, err := b.user(req.Storage, username)
```

Comment:
If `username` is always the name of the value, you can simply pass in `data` to `b.user` and have it return an error if there is a missing username. This lets you remove this check in a few other places too.